src/main/scala/org/ergoplatform/nodeView/state/UtxoStateReader.scala

```diff
@@ -38,22 +40,30 @@ trait UtxoStateReader extends ErgoStateReader with TransactionValidation {
     */
   def validateWithCost(tx: ErgoTransaction,
                        stateContextOpt: Option[ErgoStateContext],
-                       complexityLimit: Int,
+                       costLimit: Long,
                        interpreterOpt: Option[ErgoInterpreter]): Try[Long] = {
     val context = stateContextOpt.getOrElse(stateContext)
 
     val verifier = interpreterOpt.getOrElse(ErgoInterpreter(context.currentParameters))
 
+    val maxBlockCost = context.currentParameters.maxBlockCost
+    val accCost = maxBlockCost - costLimit
+
     tx.statelessValidity().flatMap { _ =>
       val boxesToSpend = tx.inputs.flatMap(i => boxById(i.boxId))
-      val txComplexity = boxesToSpend.map(_.ergoTree.complexity).sum
-      if (txComplexity > complexityLimit) {
-        throw new Exception(s"Transaction $tx has too high complexity $txComplexity")
-      }
       tx.statefulValidity(
-        boxesToSpend,
-        tx.dataInputs.flatMap(i => boxById(i.boxId)),
-        context)(verifier)
+          boxesToSpend,
+          tx.dataInputs.flatMap(i => boxById(i.boxId)),
+          context,
+          accCost)(verifier).map(_ - accCost) match {
```

Comment:
The value passed to statefulValidity shouldn't be greater than MaxTxCost limit. 
accCost, as it is computed above, is not bound by MaxTxCost.
So, changes in validateStateful are also necessary.

Reply:
Cost is bounded by MaxBlockCost, so we set accCost as there's only txCost room to reach it. Please check updated tests.